src/v3/index.ts

```diff
@@ -0,0 +1,37 @@
+import { PluginEventExtra } from '../plugin'
+import { userAgentV2 } from '../v2'
+import { detectBrowser, detectBrowserVersion, detectDevice, detectDeviceType, detectOS } from './user-agent-utils'
+
+export const userAgentV3 = (
+    event: PluginEventExtra,
+    userAgent: string,
+    global: {
+        enabledPlugin: boolean
+        enableSegmentAnalyticsJs: boolean
+        overrideUserAgentDetails: boolean
+        debugMode: boolean
+    }
+): PluginEventExtra => {
+    if (event.properties['$user_agent_plugin_disable']) {
+        return event
+    }
+
+    const vendor = event.properties['$navigator_vendor']
+
+    event.properties['$device'] = detectDevice(userAgent)
+    event.properties['$device_type'] = detectDeviceType(userAgent)
+    event.properties['$browser'] = detectBrowser(userAgent, vendor)
+    event.properties['$browser_version'] = detectBrowserVersion(userAgent, vendor)
+    const [osName, osVersion] = detectOS(userAgent)
+    event.properties['$os'] = osName
```

Comment:
Same as https://github.com/PostHog/user-agent-plugin/pull/4/files#r1522710965
We use `$os_name` on Mobile, and `$os_version`